Regression - 3.4.190 no longer works with submodule

I have a project that does not use GitVersioning itself, but submodules a project that uses GitVersioning. Using 3.4.190, it fails with an internal error "No git repo found here": 

```
error MSB4018: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task failed unexpectedly.
error MSB4018: System.ArgumentException: No git repo found here. (Parameter 'workingDirectory')
error MSB4018:    at Nerdbank.GitVersioning.Managed.ManagedGitContext..ctor(String workingDirectory, String dotGitPath, String committish)
error MSB4018:    at Nerdbank.GitVersioning.GitContext.Create(String path, String committish, Boolean writable)
error MSB4018:    at Nerdbank.GitVersioning.Tasks.GetBuildVersion.ExecuteInner()
```
I found that 3.3.37 works fine, so I used that and created a GitHub Action PR demonstrating that 3.3.37->3.4.190 upgrade breaks:

https://github.com/mhutch/MonoDevelop.Xml/actions/runs/724957601

--- a/src/NerdBank.GitVersioning/GitContext.cs
+++ b/src/NerdBank.GitVersioning/GitContext.cs
@@ -214,8 +214,16 @@ namespace Nerdbank.GitVersioning
             return committedVersion is object;
         }
 
-        private protected static bool TryFindGitPaths(string path, [NotNullWhen(true)] out string? gitDirectory, [NotNullWhen(true)] out string? workingTreeDirectory, [NotNullWhen(true)] out string? workingTreeRelativePath)
+        internal static bool TryFindGitPaths(string? path, [NotNullWhen(true)] out string? gitDirectory, [NotNullWhen(true)] out string? workingTreeDirectory, [NotNullWhen(true)] out string? workingTreeRelativePath)
         {
+            if (path is null || path.Length == 0)
+            {
+                gitDirectory = null;
+                workingTreeDirectory = null;
+                workingTreeRelativePath = null;
+                return false;
+            }
+
             path = Path.GetFullPath(path);
             var gitDirs = FindGitDir(path);
             if (gitDirs is null)

--- a/src/NerdBank.GitVersioning/ManagedGit/GitRepository.cs
+++ b/src/NerdBank.GitVersioning/ManagedGit/GitRepository.cs
@@ -40,42 +40,13 @@ namespace Nerdbank.GitVersioning.ManagedGit
         /// </returns>
         public static GitRepository? Create(string? workingDirectory)
         {
-            // Search for the top-level directory of the current git repository. This is the directory
-            // which contains a directory of file named .git.
-            // Loop until Path.GetDirectoryName returns null; in this case, we've reached the root of
-            // the file system (and we're not in a git repository).
-            while (!string.IsNullOrEmpty(workingDirectory)
-                && !File.Exists(Path.Combine(workingDirectory, GitDirectoryName))
-                && !Directory.Exists(Path.Combine(workingDirectory, GitDirectoryName)))
-            {
-                workingDirectory = Path.GetDirectoryName(workingDirectory);
-            }
-
-            if (string.IsNullOrEmpty(workingDirectory))
+            if (!GitContext.TryFindGitPaths(workingDirectory, out string? gitDirectory, out string? workingTreeDirectory, out string? workingTreeRelativePath))
             {
                 return null;
             }
 
-            var gitDirectory = Path.Combine(workingDirectory, GitDirectoryName);
-
-            if (File.Exists(gitDirectory))
-            {
-                // This is a worktree, and the path to the git directory is stored in the .git file
-                var worktreeConfig = File.ReadAllText(gitDirectory);
-
-                var gitDirStart = worktreeConfig.IndexOf("gitdir: ");
-                var gitDirEnd = worktreeConfig.IndexOf("\n", gitDirStart);
-
-                gitDirectory = worktreeConfig.Substring(gitDirStart + 8, gitDirEnd - gitDirStart - 8);
-            }
-
-            if (!Directory.Exists(gitDirectory))
-            {
-                return null;
-            }
-
-            var commonDirectory = gitDirectory;
-            var commonDirFile = Path.Combine(gitDirectory, "commondir");
+            string commonDirectory = gitDirectory;
+            string commonDirFile = Path.Combine(gitDirectory, "commondir");
 
             if (File.Exists(commonDirFile))
             {
@@ -83,7 +54,7 @@ namespace Nerdbank.GitVersioning.ManagedGit
                 commonDirectory = Path.Combine(gitDirectory, commonDirectoryRelativePath);
             }
 
-            var objectDirectory = Path.Combine(commonDirectory, "objects");
+            string objectDirectory = Path.Combine(commonDirectory, "objects");
 
             return new GitRepository(workingDirectory!, gitDirectory, commonDirectory, objectDirectory);
         }

